src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/ArtifactsParametersRuleTests.cs

```diff
@@ -0,0 +1,498 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Linter.Rules;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests
+{
+    [TestClass]
+    public class ArtifactsParametersRuleTests : LinterRuleTestsBase
+    {
+        private void CompileAndTest(string bicepText, string[] expectedMessagesForCode, OnCompileErrors onCompileErrors = OnCompileErrors.Fail)
+        {
+            base.AssertLinterRuleDiagnostics(ArtifactsParametersRule.Code, bicepText, expectedMessagesForCode, onCompileErrors);
+        }
+
+        [TestMethod]
+        public void MissingUnderscoreInParamNames_ArtifactsLocation()
+        {
+            CompileAndTest(
+                @"
+                    @secure()
+                    param artifactsLocation string = 'something'
+
+                    @secure()
+                    param _artifactsLocationSasToken string = ''
+                ",
+                new string[]
+                {
+                    "The artifacts parameter '_artifactsLocation' must start with an underscore.",
+                    "If the '_artifactsLocation' parameter has a default value, it must be a raw URL or an expression like 'deployment().properties.templateLink.uri'.",
+                }
+           );
+        }
+
+        [TestMethod]
+        public void MissingUnderscoreInParamNames_CaseInsensitive()
+        {
+            CompileAndTest(
+                @"
+                    @secure()
+                    param artifactsLOCATION string = 'something'
+
+                    @secure()
+                    param _ARTIFACTSLocationSasToken string = ''
+                ",
+                new string[]
+                {
+                    "The artifacts parameter '_artifactsLocation' must start with an underscore.",
+                    "If the '_artifactsLocation' parameter has a default value, it must be a raw URL or an expression like 'deployment().properties.templateLink.uri'.",
+                }
+           );
+        }
+
+        [TestMethod]
+        public void MissingUnderscoreInParamNames_SasToken()
+        {
+            CompileAndTest(
+                @"
+                    @secure()
+                    param _artifactsLocation string = deployment().properties.templateLink.uri
+
+                    @secure()
+                    param artifactsLocationSasToken string = 'something'
+                ",
+                new string[]
+                {
+                    "The artifacts parameter '_artifactsLocationSasToken' must start with an underscore.",
```

Comment:
Same as above. Should be 'artifactsLocationSasToken' instead of '_artifactsLocationSasToken'?